File: 프로그래머스_고득점kit/Stack&Queue/같은숫자는싫어.py
# stack의 top끼리 비교해 다르면 push
def solution(arr):
    answer = []
    last = -1 # java -> Arraylist의 top 구하기 번거로워 이 방식
    for el in arr:
        if last != el:
            answer.append(el)
            last = el
    return answer

def solution(arr): # java -> Stack<Integer> 사용
    answer = []
    for el in arr:
        if len(answer) == 0 or el != answer[-1]:
            answer.append(el)
    return answer

File: 프로그래머스_고득점kit/Stack&Queue/같은숫자는싫어.java
import java.util.*;

class Solution {
    public int[] solution(int []arr) {
        List<Integer> tempList = new ArrayList<Integer>();
        int last = -1;
        for (int el : arr) {
            if (last != el) {
                tempList.add(el);
                last = el;
            }
        }

        int[] answer = new int[tempList.size()];
        for (int i = 0 ; i < answer.length ; i++) {
            answer[i] = tempList.get(i);
        }
        return answer;
    }

    public int[] solution2(int []arr) {
        Stack<Integer> stack = new Stack<Integer>();
        for (int el : arr) {
            if (stack.empty() || stack.peek() != el) {
                stack.add(el);
            }
        }

        int[] answer = new int[stack.size()];
        for (int i = 0 ; i < answer.length ; i++) {
            answer[i] = stack.get(i);
        }
        return answer;
    }
}